academic_observatory_workflows/workflows/dqc_workflow.py

```diff
@@ -0,0 +1,456 @@
+                            mode="reschedule",
+                            duration=timedelta(days=7),
+                            poke_interval=int(timedelta(hours=1).total_seconds()),
+                            timeout=int(timedelta(days=3).total_seconds()),
+                        )
+                        print(sensor)
+                        self.add_operator(sensor)
+                        task_sensors.append(sensor)
+
+                    # Add all of the sesnors to tg_sensors with this line.
+                    (task_sensors)
+
+                    task_sensor_group = tg_sensors
+
+            # Add the standard tasks for the workflow
+            # fmt: off
+            task_check_dependencies = PythonOperator(python_callable=self.check_dependencies, task_id="check_dependencies")
+            task_create_dataset = self.make_python_operator(self.create_dataset, "create_dataset")
+            # fmt: on
+
+            # Add each dataset as a task group and perform the checks on the tables in parallel.
+            task_datasets_group = []
+            for dataset_id in list(self.datasets.keys()):
+                with TaskGroup(group_id=dataset_id) as tg_dataset:
+                    tables: List[Table] = make_tables(self.project_id, dataset_id, self.datasets[dataset_id]["tables"])
+
+                    table_tasks = []
+                    for table in tables:
+                        task = self.make_python_operator(
+                            self.perform_data_quality_check,
+                            task_id=f"{table.table_id}",
+                            op_kwargs={f"task_id": f"{table.table_id}", "table": table},
+                        )
+                        table_tasks.append(task)
+
+                    # Add the list of table_tasks to tg_dataset by this line
+                    (table_tasks)
+
+                    task_datasets_group.append(tg_dataset)
+
+            # Link all tasks and task groups together.
+            (task_sensor_group >> task_check_dependencies >> task_create_dataset >> task_datasets_group)
+
+        return self.dag
+
+    def make_release(self, **kwargs) -> Release:
+        """Make a release instance.
+
+        :param kwargs: the context passed from the PythonOperator. See
+        https://airflow.apache.org/docs/stable/macros-ref.html for a list of the keyword arguments that are passed
+        to this argument.
+        :return: A release instance.
+        """
+        return Release(dag_id=self.dag_id, run_id=kwargs["run_id"])
+
+    def create_dataset(self, release: Release, **kwargs):
+        """Create a dataset for all of the DQC tables for the workflows."""
+
+        success = bq_create_dataset(
+            project_id=self.project_id,
+            dataset_id=self.bq_dataset_id,
+            location=self.data_location,
+            description=self.bq_dataset_description,
+        )
+
+        set_task_state(success, self.create_dataset.__name__, release)
```

Comment:
This function doesn't return success, I think it returns the dataset and may throw an exception if it can't create the dataset? So you can probably remove success and set task state.